src/pages/projectSettings/CopyProjectModal.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from "react";
+import { useMutation } from "@apollo/client";
+import { useHistory } from "react-router-dom";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import { SpruceForm } from "components/SpruceForm";
+import { getProjectSettingsRoute } from "constants/routes";
+import { useToastContext } from "context/toast";
+import {
+  CopyProjectMutation,
+  CopyProjectMutationVariables,
+} from "gql/generated/types";
+import { COPY_PROJECT } from "gql/mutations";
+import { projectId, projectName } from "./sharedFormSchema";
+
+interface Props {
+  handleClose: () => void;
+  id: string;
+  label: string;
+  open: boolean;
+}
+
+export const CopyProjectModal: React.VFC<Props> = ({
+  handleClose,
+  id,
+  label,
+  open,
+}) => {
+  const dispatchToast = useToastContext();
+  const { replace } = useHistory();
+
+  const [formState, setFormState] = useState({
+    projectId: "",
+    projectName: "",
+  });
+  const [hasError, setHasError] = useState(true);
+
+  const [copyProject] = useMutation<
+    CopyProjectMutation,
+    CopyProjectMutationVariables
+  >(COPY_PROJECT, {
+    onCompleted({ copyProject: { identifier } }) {
+      dispatchToast.success(`Successfully created the project: ${identifier}`);
+      replace(getProjectSettingsRoute(identifier));
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error creating the project: ${err.message}`
```

Comment:
For many errors, we still create the project just some things may not be carried over. For example, 
<img width="438" alt="image" src="https://user-images.githubusercontent.com/26798134/165820222-dd1a9cd4-15d8-460b-92c9-6440062d3bf3.png">

When copying the spruce project I get errors about admins, however [at that point in the code|https://github.com/evergreen-ci/evergreen/blob/df23005997ff5ce934cc6114ae353a27ea42ab3e/rest/data/project_settings.go#L76] we've already copied everything else. Should we still redirect then and indicate t's a partial error? (We could modify the copy function to still return the item and/or try all of the pieces and return the entire error if that's better)

Reply:
Discussed offline, this will require additional back and front end work. EVG-16784 has been created to capture these changes.